Fixes to COM wrappers

kant2002 opened the pull request:
- Create IUnknown with 1 as ref count
- Add error message when cannot QI to desired interface

jkotas commented on src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs:
```diff
@@ -506,9 +506,11 @@ internal static unsafe IntPtr ComInterfaceForObject(object instance, Guid target
             int resultCode = wrapper->QueryInterface(in targetIID, out comObjectInterface);
             if (resultCode != 0)
             {
```
This path leaks the `unknownPtr` reference.

jkotas commented on src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs:
```diff
@@ -506,9 +506,11 @@ internal static unsafe IntPtr ComInterfaceForObject(object instance, Guid target
             if (resultCode != 0)
             {
-                throw new PlatformNotSupportedException(SR.PlatformNotSupported_ComInterop);
+                throw new PlatformNotSupportedException(SR.InvalidOperation_InterfaceNotSupported);
```
Is this the exception that CoreCLR throws in this situation? (I would expect it to be something else.)

kant2002 replied:
Cannot find location where it happens in CoreCLR. Will try again when would be home in hour or so. Will look at marshalling code.

kant2002 replied:
Is this proper location where look for object -> interface conversion.

https://github.com/dotnet/runtime/blob/aaaaeed8c8d2d01dccd1f7fc536ea27eee75f472/src/coreclr/vm/ilmarshalers.cpp#L1401

Maybe Marshal class will throw this exception too.
I account for case where ComWrappers does not provide correct interface vtbl

jkotas replied:
The easiest way to find out is to write a small test.

jkotas commented on src/coreclr/nativeaot/System.Private.CoreLib/src/Resources/Strings.resx:
```diff
@@ -3160,4 +3160,7 @@
   <data name="InvalidOperation_ComInteropRequireComWrapperInstance" xml:space="preserve">
     <value>COM Interop requires ComWrapper instance registered for marshalling.</value>
   </data>
+  <data name="InvalidOperation_InterfaceNotSupported" xml:space="preserve">
```
Could you please revert all changes in this file?

kant2002 replied:
@jkotas done

kant2002 committed:
Fixes to COM wrappers
- Create IUnknown with 1 as ref count
- Add error message when cannot QI to desired interface

jkotas reviewed (approved):
Thank you!
